### PR TITLE
[Fixes #146] Maximum Key Length on Primary Key

### LINKED ISSUE #146: Maximum Key Length on Primary Key
The ID on the caching table is nvarchar(900); with a maximum key length of 900 bytes won't the ID column have a max size of 1800 byes?

When executing the create table statement, SQL Server does generate a warning.
`Warning! The maximum key length is 900 bytes. The index 'pk_Id' has maximum length of 1800 bytes. For some combination of large values, the insert/update operation will fail.`

### PR DESCRIPTION
**Pull request by kichalla**
@divega @muratg

### CODE REVIEW

File: src/dotnet-sql-cache/SqlQueries.cs [hunk 1]
```diff
@@ -8,8 +8,10 @@ namespace Microsoft.Extensions.Caching.SqlConfig
     internal class SqlQueries
     {
         private const string CreateTableFormat = "CREATE TABLE {0}(" +
-            // add collation to the key column to make it case-sensitive
-            "Id nvarchar(900) COLLATE SQL_Latin1_General_CP1_CS_AS NOT NULL, " +
+            // - Primary key column cannot exceed the maximum size of 900 bytes. Here each character occupies
+            //   2 bytes.
+            // - Add collation to the key column to make it case-sensitive
+            "Id nvarchar(445) COLLATE SQL_Latin1_General_CP1_CS_AS NOT NULL, " +
```
Comment (kichalla) [inline src/dotnet-sql-cache/SqlQueries.cs]: From https://msdn.microsoft.com/en-us/library/ms186939.aspx, the storage size is the (size * 2 bytes) + additional 2 bytes. So this could be just 449 but not sure if we really care it to be that accurate.

File: samples/SqlServerCacheConcurencySample/project.json [hunk 2]
```diff
@@ -11,6 +11,10 @@
     "Microsoft.Extensions.PlatformAbstractions": "1.0.0-*"
   },
```
Comment (muratg) [inline samples/SqlServerCacheConcurencySample/project.json]: Can you make it consistent with the others? (You removed an empty line from the other project.json below.)

File: src/dotnet-sql-cache/SqlQueries.cs (hunk 1 above)
Comment (muratg) [inline src/dotnet-sql-cache/SqlQueries.cs]: If we know the exact number, any concerns just providing that?

Comment (divega) [inline src/dotnet-sql-cache/SqlQueries.cs]: I agree with @muratg. These specs are pretty stable and there doesn't seem to be a downside on allowing the maximum key length possible. Maybe add tests that exercise the maximum size and a negative test that exceeds it (do we know what happens when you do?).

**Comment by kichalla** [comment]
:up: :date: Adding tests

File: src/dotnet-sql-cache/SqlQueries.cs (hunk 1 above)
Comment (kichalla) [inline src/dotnet-sql-cache/SqlQueries.cs]: > do we know what happens when you do?
I noticed that if a key greater than 898 bytes is provided, then the value is truncated to 898.

**Comment by muratg** [comment]
:shipit:
